### PR TITLE
Knative build for in-cluster image builds in Kubeflow

### PR DESCRIPTION
**Pull request by swiftdiaries**
Addresses issue: https://github.com/kubeflow/kubeflow/issues/1317

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/1804)

### CODE REVIEW

**Comment by abhi-g** [comment]
/assign @lluunn 
Do you mind reviewing this PR. Please reassign if you think someone else should look at it.

**Comment by lluunn** [comment]
Do you need Istio?
https://github.com/knative/docs/blob/master/install/Knative-with-GKE.md#installing-istio

Or knative build itself doesn't depend on istio?

File: kubeflow/knative-build/knative-build.libsonnet [hunk 1]
```diff
@@ -0,0 +1,379 @@
+{
```
Comment (lluunn) [inline kubeflow/knative-build/knative-build.libsonnet]: Is this converted from https://raw.githubusercontent.com/knative/serving/v0.1.1/third_party/config/build/release.yaml?
anything changed?

**Comment by swiftdiaries** [comment]
It doesn’t depend on Istio.
Yes, nothing was changed.

**Comment by lluunn** [comment]
Thanks.
/lgtm
/approve

**Pull request merged**